tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -953,12 +957,16 @@ def _train_and_save(self, X, y, model: AbstractModel, X_val=None, y_val=None, st
                 if time_limit <= 0:
                     logger.log(15, f'Skipping {model.name} due to lack of time remaining.')
                     return model_names_trained
+                if pruned_features and time_limit and model.is_fit() and model.fit_time > time_limit:
+                    logger.log(15, f'Skipping feature pruning for {model.name} due to projected lack of time.')
```

Comment:
I'd make this logging statement level 20, and have your message additionally clarify to the user: 
"Consider increasing `time_limit` to train this model with feature pruning"